### PR TITLE
fix(heder-filters): fix position in one row on tablet and mobile

### PR DESCRIPTION
**Pull request by dmitryuck**

### CODE REVIEW

File: src/shared/countries-filter/countries-filter.component.ts [hunk 1]
```diff
@@ -397,9 +444,13 @@ export class CountriesFilterComponent implements OnInit, OnDestroy, OnChanges {
 
     if (regions[0] === 'World' && countries[0] !== 'World') {
       if (countries.length > 2) {
-        this.activeCountries = getTranslatedCountries.slice(0, 2).join(', ') + ' (+' + (getTranslatedCountries.length - 2) + ')';
+        this.activeCountries = getTranslatedCountries.slice(0, this.sliceCount).join(', ') + ' (+' + (getTranslatedCountries.length - this.sliceCount) + ')';
```
Comment (rychkog) [inline src/shared/countries-filter/countries-filter.component.ts]: Use template strings syntax

File: src/shared/countries-filter/countries-filter.component.ts [hunk 2]
```diff
@@ -397,9 +444,13 @@ export class CountriesFilterComponent implements OnInit, OnDestroy, OnChanges {
...
       } else {
-        this.activeCountries = getTranslatedCountries.join(' & ');
+        if (this.sliceCount === 1 && countries.length > 1) {
+          this.activeCountries = getTranslatedCountries.slice(0, this.sliceCount).join(', ') + ' (+' + (getTranslatedCountries.length - this.sliceCount) + ')';
```
Comment (rychkog) [inline src/shared/countries-filter/countries-filter.component.ts]: Use template strings syntax

File: src/shared/countries-filter/countries-filter.component.ts [hunk 3]
```diff
@@ -412,7 +463,7 @@ export class CountriesFilterComponent implements OnInit, OnDestroy, OnChanges {
 
     if (regions[0] !== 'World') {
       if (regions.length > 2) {
-        this.activeCountries = getTranslatedCountries.slice(0, 2).join(', ') + ' (+' + (getTranslatedCountries.length - 2) + ')';
+        this.activeCountries = getTranslatedCountries.slice(0, this.sliceCount).join(', ') + ' (+' + (getTranslatedCountries.length - this.sliceCount) + ')';
```
Comment (rychkog) [inline src/shared/countries-filter/countries-filter.component.ts]: Use template strings syntax

File: src/shared/countries-filter/countries-filter.component.ts [hunk 4]
```diff
@@ -443,7 +494,7 @@ export class CountriesFilterComponent implements OnInit, OnDestroy, OnChanges {
 
           this.activeCountries = difference.length === 1 && regions.length === 1 ?
                                 getTranslatedRegions[0] + ' & ' + difference[0] :
-                                getTranslatedCountries.slice(0, 2).join(', ') + ' (+' + (activeCountriesNum) + ')';
+                                getTranslatedCountries.slice(0, this.sliceCount).join(', ') + ' (+' + (activeCountriesNum - this.sliceCount) + ')';
```
Comment (rychkog) [inline src/shared/countries-filter/countries-filter.component.ts]: Use template strings syntax

File: src/shared/countries-filter/countries-filter.component.ts [hunk 5]
```diff
@@ -470,7 +521,7 @@ export class CountriesFilterComponent implements OnInit, OnDestroy, OnChanges {
     let concatLocations: string[] = regions.concat(getTranslatedCountries);
 
     if (concatLocations.length > 2) {
-      this.activeCountries = concatLocations.slice(0, 2).join(', ') + ' (+' + (concatLocations.length - 2) + ')';
+      this.activeCountries = concatLocations.slice(0, this.sliceCount).join(', ') + ' (+' + (concatLocations.length - this.sliceCount) + ')';
```
Comment (rychkog) [inline src/shared/countries-filter/countries-filter.component.ts]: Use template strings syntax

File: src/shared/header/header.component.ts [hunk 6]
```diff
@@ -60,18 +63,24 @@ export class HeaderComponent implements OnInit, OnChanges, OnDestroy, AfterViewI
   public device: BrowserDetectionService;
   public isDesktop: boolean;
   public isMobile: boolean;
+  public isTablet: boolean;
   public languageService: LanguageService;
   public getTranslationSubscribe: Subscription;
+  public resizeSubscribe: Subscription;
```
Comment (rychkog) [inline src/shared/header/header.component.ts]: resizeSubscription

Comment (rychkog) [reply src/shared/header/header.component.ts]: orientationChangeSubscription

File: src/shared/header/header.component.ts [hunk 7]
```diff
@@ -84,12 +93,16 @@ export class HeaderComponent implements OnInit, OnChanges, OnDestroy, AfterViewI
     this.mapComponent = this.activatedRoute.snapshot.url[0].path === 'map';
   }
 
-  /* tslint:disable-next-line */
-  public ngAfterViewInit(): void {}
+  public ngAfterViewInit(): void {
+    this.incomeContainer = this.element.querySelector('.income-title-container') as HTMLElement;
```
Comment (rychkog) [inline src/shared/header/header.component.ts]: Could you get rid of this query selector and use @ViewChild?

File: src/shared/header/header.component.ts [hunk 8]
```diff
@@ -101,12 +114,32 @@ export class HeaderComponent implements OnInit, OnChanges, OnDestroy, AfterViewI
 
         this.streetData = res.data;
       });
+
+    this.resizeSubscribe = fromEvent(window, 'resize')
+      .debounceTime(150)
+      .subscribe(() => {
+        this.zone.run(() => {
```
Comment (rychkog) [inline src/shared/header/header.component.ts]: Are you sure you need a zone here?

**Review by rychkog — CHANGES_REQUESTED** [review]

File: src/shared/header/header.component.ts (hunk 7 above)
Comment (dmitryuck) [reply src/shared/header/header.component.ts]: We have a hole section in performance iteration to replace querySelectors with ViewChild let do this during it

File: .config/tools/bstack-local.js [hunk 9]
```diff
@@ -9,13 +9,13 @@ const BrowserStackTunnel = require('browserstacktunnel-wrapper');
 const cwd = path.dirname(require.main.filename);
 
 const confDir = path.dirname(cwd);
-const bsConfig = require(path.dirname(confDir)+'/bs-config.json');
+// const bsConfig = require(path.dirname(confDir)+'/bs-config.json');
```
Comment (rychkog) [inline .config/tools/bstack-local.js]: Why is it here? If it is not needed anymore - should be removed

File: .config/tools/bstack-local.js [hunk 10]
```diff
@@ -9,13 +9,13 @@ const BrowserStackTunnel = require('browserstacktunnel-wrapper');
...
 
 const TUNNEL_IDENTIFIER = process.env.TRAVIS_BUILD_NUMBER;
 const READY_FILE = path.join(' ', 'tmp', 'bstack-' + TUNNEL_IDENTIFIER + '.ready').trim();
 const ACCESS_KEY = process.env.BROWSER_STACK_ACCESS_KEY;
 const HOST = 'localhost';
-const PORT = process.env.PORT || bsConfig.port;
+const PORT = process.env.PORT; //|| bsConfig.port;
```
Comment (rychkog) [inline .config/tools/bstack-local.js]: same as above

File: src/shared/countries-filter/countries-filter.component.ts [hunk 11]
```diff
@@ -22,10 +22,19 @@ import { BrowserDetectionService, CountriesFilterService, LanguageService } from
   styleUrls: ['./countries-filter-mobile/countries-filter-mobile.component.css', './countries-filter.component.css']
 })
 export class CountriesFilterComponent implements OnInit, OnDestroy, OnChanges {
+  @Input()
+  public url: string;
+
+  @Output('isFilterGotData')
```
Comment (rychkog) [inline src/shared/countries-filter/countries-filter.component.ts]: you don't need to specify name of the output if it is the same as field name

File: src/shared/countries-filter/countries-filter.component.ts [hunk 12]
```diff
@@ -22,10 +22,19 @@ import { BrowserDetectionService, CountriesFilterService, LanguageService } from
...
+  public isFilterGotData: EventEmitter<any> = new EventEmitter<any>();
+  @Output()
+  public selectedFilter: EventEmitter<any> = new EventEmitter<any>();
```
Comment (rychkog) [inline src/shared/countries-filter/countries-filter.component.ts]: any ?

**Review by rychkog — CHANGES_REQUESTED** [review]

**Pull request merged**